File: Excel/900-999/991/Challenge 991.R
library(tidyverse)
library(readxl)

path <- "900-999/991/991 Opposite Directions Removal.xlsx"
input <- read_excel(path, range = "A1:B21")
test <- read_excel(path, range = "C1:C21") %>%
  mutate(across(everything(), ~ replace_na(as.character(.x), "")))

remove_opposite_directions <- function(directions) {
  s <- gsub("[, ]", "", directions)
  repeat {
    t <- gsub("NS|SN|EW|WE", "", s)
    if (t == s) {
      break
    }
    s <- t
  }
  paste(str_split(s, "")[[1]], collapse = ",")
}

result <- map_chr(input$`Path`, remove_opposite_directions)
all.equal(test$`Answer Expected`, result)
# [1] TRUE

File: Excel/900-999/991/Challenge 991.py
import pandas as pd
import re

path = "900-999/991/991 Opposite Directions Removal.xlsx"
input = pd.read_excel(path, usecols="A:B", nrows=21, skiprows=0)
test = pd.read_excel(path, usecols="C:C", nrows=21, skiprows=0).fillna("")


def remove_opposite_directions(directions):
	s = re.sub(r"[, ]", "", directions)
	while True:
		t = re.sub(r"NS|SN|EW|WE", "", s)
		if t == s:
			break
		s = t
	return ",".join(list(s))

input["Output"] = input["Path"].apply(remove_opposite_directions)
print(input['Output'].equals(test['Answer Expected']))
# True